src/win/fs.c

```diff
@@ -1941,8 +1985,15 @@ static void fs__fstat(uv_fs_t* req) {
   }
 
   if (fs__stat_handle(handle, &req->statbuf, 0) != 0) {
-    SET_REQ_WIN32_ERROR(req, GetLastError());
-    return;
+    if (!fs__is_stdio_fd(fd)) {
+      SET_REQ_WIN32_ERROR(req, GetLastError());
+      return;
+    }
+
+    if (fs__stat_stdio_fd(fd, &req->statbuf) != 0) {
+      SET_REQ_WIN32_ERROR(req, ERROR_INVALID_FUNCTION);
```

Comment:
Why ERROR_INVALID_FUNCTION? Libuv remaps it to UV_EISDIR but that doesn't seem appropriate here.

Reply:
Mainly for consistency with previous behavior. This is the same error that was returned once `fs__stat_handle` failed (although I agree, it's not very appropriate). Another thing to keep in mind here is that the only way for `fs__stat_stdio_fd` to fail, would be for `fstat` to fail and based on [docs](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/fstat-fstat32-fstat64-fstati64-fstat32i64-fstat64i32?view=msvc-170), this should never happen since we verify file descriptor beforehand and stat buffer is never NULL in our case.

Reply:
@bnoordhuis do you have some suggestion for a more meaningful error code?